examples/utils_rf.py

```diff
@@ -0,0 +1,131 @@
+    import numpy as np
+    lims = [
+        np.min([ax.get_xlim(), ax.get_ylim()]),  # min of both axes
+        np.max([ax.get_xlim(), ax.get_ylim()]),  # max of both axes
+    ]
+
+    # now plot both limits against eachother
+    ax.plot(lims, lims, 'k-', alpha=0.75, zorder=0)
+    ax.set_xlim(lims)
+    ax.set_ylim(lims)
+    
+def prettify_plot(ax):
+    """utility function for making plots prettier"""
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    
+    
+def plot_results(ax, truth, pred, title):
+    from sklearn.metrics import median_absolute_error, r2_score, mean_squared_error
+    from sklearn.model_selection import train_test_split
+    ax.scatter(y=pred, x=truth, zorder=2.6, s=20, alpha=0.5)
+    props = {'boxstyle':'round', 'facecolor':'wheat'}
+    ax.text(x=0.05, y=0.8, transform=ax.transAxes,
+            s=f'$R^2$={np.round(r2_score(pred, truth),2)}, ' +\
+                f'MAE={np.round(median_absolute_error(pred, truth),3)}, ' +\
+                f'RMSE={np.round( np.sqrt(mean_squared_error(pred, truth)), 2)}',
+            bbox=props)
+
+    # Change axis formatting
+    prettify_plot(ax)
+    add_121_line(ax)
+    ax.set_title(title)
+    ax.set_xlabel('Truth')
+    ax.set_ylabel('Prediction')
+    
+    
+def LeaveOneOut(Xdata, Ydata, rndseed=0, **rf_kwargs):
+    """
+    Function to perform LeaveOneOut cross-validation with different estimators. 
+    """
+    from GCEm.rf_model import RFModel
+    
+    # fix random seed for reproducibility 
+    # then shuffle X,Y indices so that they're not ordered (in time, for example)
+    np.random.seed(rndseed)
+    rndperm = np.random.permutation(Xdata.shape[0])
+    
+    # How many indices?
+    n_data = Xdata.shape[0]
+    
+    # Output array for test val, pred_mean
+    output = np.vstack([np.empty(2) for _ in range(n_data)])
+    
+    for test_idx in range(n_data):
+        """Split into training - validation sets"""
+        X_train = Xdata.iloc[rndperm[np.arange(len(rndperm))!= test_idx], :]
+        X_test  = Xdata.iloc[rndperm[test_idx], :].to_numpy().reshape(1,-1)
+        
+        Y_train = Ydata[rndperm[np.arange(len(rndperm))!= test_idx]]
+        Y_test  = Ydata[rndperm[test_idx]]
+       
+        """Construct and fit model"""
+        model = RFModel(training_params=X_train, 
+                        training_data=Y_train, 
+                        random_state=rndseed, 
+                        **rf_kwargs)
+        
+        model.train()
+            
+        """Evaluate model on test data"""
+        predictions,v = model.predict(X_test)
+        
+        """Save output for validation plot  later on"""
+        output[test_idx] = (Y_test, predictions)
+
+    return output
+
+def get_crm_data(cache_path='.', preprocess=True):
```

Comment:
I think this is probably the only function that belongs in /examples, could you merge this one with examples/utils.py and the rest with GCEm/utils.py

Reply:
Done!